library/VTKExtensions/Readers/vtkF3DMemoryMesh.cxx

```diff
@@ -0,0 +1,64 @@
+#include "vtkF3DMemoryMesh.h"
+
+#include "vtkFloatArray.h"
+#include "vtkInformationVector.h"
+#include "vtkObjectFactory.h"
+#include "vtkPointData.h"
+
+vtkStandardNewMacro(vtkF3DMemoryMesh);
+
+//------------------------------------------------------------------------------
+vtkF3DMemoryMesh::vtkF3DMemoryMesh()
+{
+  this->SetNumberOfInputPorts(0);
+}
+
+//------------------------------------------------------------------------------
+vtkF3DMemoryMesh::~vtkF3DMemoryMesh() = default;
+
+//------------------------------------------------------------------------------
+void vtkF3DMemoryMesh::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os, indent);
+}
+
+//------------------------------------------------------------------------------
+void vtkF3DMemoryMesh::SetPoints(const std::vector<float>& positions)
+{
+  vtkNew<vtkPoints> points;
+  points->SetNumberOfPoints(positions.size() / 3);
+
+  for (size_t i = 0; i < positions.size() / 3; i++)
+  {
+    points->SetPoint(i, &positions[3 * i]);
+  }
```

Comment:
you could access the internal array and use ranges to improve performance when TBB is enabled

Reply:
can you give me an example of ranges usage?

Reply:
I would need to spend more time on this if you need an actual snippet of code but should be something based on:

C++11 for-range loops with vtk::DataArrayTupleRange : https://www.kitware.com/c11-for-range-support-in-vtk/

and

vtKSMP::Transform: https://vtk.org/doc/nightly/html/classvtkSMPTools.html#a16e88039a1e33aaacef235e92779c4ee

Reply:
ok I'll figure out